lgc/state/PalMetadata.cpp

```diff
@@ -642,3 +695,58 @@ void PalMetadata::getVsEntryRegInfo(VsEntryRegInfo &regInfo) {
   if (m_pipelineState->getTargetInfo().getGfxIpVersion().major < 10)
     regInfo.wave32 = false;
 }
+
+// =====================================================================================================================
+// Get the llvm type for that corresponds to tyName.  Returns nullptr if no such type exists.
+//
+// @param tyName : A type name that is an encoding of a type.
+llvm::Type *PalMetadata::getLlvmType(StringRef tyName) const {
+  unsigned vecLength = 0;
+  if (tyName[0] == 'v') {
+    tyName = tyName.drop_front();
+    tyName.consumeInteger(10, vecLength);
+  }
+  Type *ty = nullptr;
+  if (tyName == "i8")
+    ty = Type::getInt8Ty(m_pipelineState->getContext());
+  else if (tyName == "i16")
+    ty = Type::getInt16Ty(m_pipelineState->getContext());
+  else if (tyName == "i32")
+    ty = Type::getInt32Ty(m_pipelineState->getContext());
+  else if (tyName == "i64")
+    ty = Type::getInt64Ty(m_pipelineState->getContext());
+  else if (tyName == "f16")
+    ty = Type::getHalfTy(m_pipelineState->getContext());
+  else if (tyName == "f32")
+    ty = Type::getFloatTy(m_pipelineState->getContext());
+  else if (tyName == "f64")
+    ty = Type::getDoubleTy(m_pipelineState->getContext());
+  if (vecLength != 0 && ty != nullptr)
+    ty = FixedVectorType::get(ty, vecLength);
+  return ty;
+}
+
+// =====================================================================================================================
+// Updates the SPI_SHADER_COL_FORMAT entry.
+//
+void PalMetadata::updateSpiShaderColFormat(const SmallVector<ColorExportInfo, 8> &exps) {
+  unsigned spiShaderColFormat = 0;
+  for (auto &exp : exps) {
+    unsigned expFormat = m_pipelineState->computeExportFormat(exp.ty, exp.location);
+    spiShaderColFormat |= (expFormat << (4 * exp.hwColorTarget));
+  }
+
+  unsigned depthExpFmt = getRegister(mmSPI_SHADER_Z_FORMAT);
```

Comment:
Why do we need to read previous values of these registers? Where would they have come from?

Reply:
Originally I wrote this to only be called while linking, so depth export format would have been written [here](https://github.com/GPUOpen-Drivers/llpc/blob/dev/lgc/patch/Gfx9ConfigBuilder.cpp#L1856).  That is no longer true.  I have to change this.